### PR TITLE
Added regex parser

### PR DESCRIPTION
**Pull request by maniodev**
Following the issue https://github.com/joshbuddy/jsonpath/issues/74, there is a fix proposal.
I used `to_regexp` gem to make things easier.

### CODE REVIEW

File: jsonpath.gemspec [hunk 1]
```diff
@@ -29,5 +29,6 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'minitest', '~> 2.2.0'
   s.add_development_dependency 'phocus'
   s.add_development_dependency 'bundler'
+  s.add_development_dependency 'to_regexp'
```
Comment (Skarlso) [inline jsonpath.gemspec]: Please add it as a full dependency since it's not only a development feature.

**Comment by Skarlso** [comment]
Awesome. Will merge and release soon.

**Comment by maniodev** [comment]
Thanks :)

**Comment by Skarlso** [comment]
Released.

**Pull request merged**